### PR TITLE
Fix fetching tx fee

### PR DESCRIPTION
**Pull request by pawell24**
**Adapting UI to changes in SDK regarding network fee calculation, improving UX related to route fetching.**

## Summary by CodeRabbit

- **New Features**
	- Enhanced fee estimation process with improved error handling and state management.
	- Introduced visual feedback for loading states with an animated logo.
	- Added new screen states: `LOADING`, `CLAIM`, and `NOT_FOUND` for better user interface management.

- **Bug Fixes**
	- Improved handling of fee estimation errors and state transitions during the transaction lifecycle.
  
- **Dependency Updates**
	- Updated the `@squirrel-labs/peanut-sdk` version from `^0.5.0` to `^0.5.2`, potentially introducing new features and improvements.

### CODE REVIEW

File: src/components/Request/Pay/utils.ts [hunk 1]
```diff
@@ -10,3 +10,9 @@ export enum EPeanutLinkType {
     native,
     erc20,
 }
+
+export enum RequestScreens {
+    LOADING = 'LOADING',
+    CLAIM = 'CLAIM',
+    NOT_FOUND = 'NOT_FOUND',
+}
```
Comment (Hugo0) [inline src/components/Request/Pay/utils.ts]: 1. Is this the same enum we have in claim screens?
2. Are these all the possible states?
3. Is RequestScreens the correct name for an enum of strings?

File: src/components/Request/Pay/Views/Initial.view.tsx [hunk 2]
```diff
@@ -52,19 +54,36 @@ export const InitialView = ({
...
-                unsignedTxs: xchainUnsignedTxs.unsignedTxs,
-                isNativeTxValue: selectedTokenAddress === ADDRESS_ZERO ? true : false,
-                fromAmount: requestLinkData.tokenAmount,
-            })
-
-            const tokenPriceData = await utils.fetchTokenPrice(ADDRESS_ZERO, selectedChainID)
-            const price = tokenPriceData?.price ?? 0
-            setTxFee((Number(ethers.utils.formatEther(txFee)) * price).toFixed(2))
+            setLinkState(RequestScreens.LOADING)
+            if(selectedChainID === requestLinkData.chainId && selectedTokenAddress === requestLinkData.tokenAddress) {
```
Comment (Hugo0) [inline src/components/Request/Pay/Views/Initial.view.tsx]: Do we have a reset to origin chain option (like in x-chain claim) ?

**Review by Hugo0 — CHANGES_REQUESTED** [review]

File: src/components/Request/Pay/utils.ts (hunk 1 above)
Comment (pawell24) [reply src/components/Request/Pay/utils.ts]: 1. It's a bit different, this enum, checks the status of the fetching route
2. Yes, all states are possible
3. I changed name of this enum

File: src/components/Request/Pay/Views/Initial.view.tsx (hunk 2 above)
Comment (pawell24) [reply src/components/Request/Pay/Views/Initial.view.tsx]: I don't understand, can you describe in more detail what this means?

**Pull request merged**